File: Android_app/pioneer/src/main/java/com/ABC/pioneer/sensor/Device.java
package com.ABC.pioneer.sensor;

import com.ABC.pioneer.sensor.datatype.TargetIdentifier;

import java.util.Date;

public class Device {
    ///设备注册时间戳
    public final Date createdAt;
    ///上次任何更改，例如attribute更新
    public Date lastUpdatedAt = null;
    //临时设备标识符，例如 外设标识符UUID
    public final TargetIdentifier identifier;

    public Device(TargetIdentifier identifier) {
        this.createdAt = new Date();
        this.lastUpdatedAt = this.createdAt;
        this.identifier = identifier;
    }

    public Device(Device device, TargetIdentifier identifier) {
        this.createdAt = device.createdAt;
        this.lastUpdatedAt = new Date();
        this.identifier = identifier;
    }
}

File: IOS_app/Pioneer_ios/Pioneer Framework/Pioneer/Sensor/Device.swift
//
//  Device.swift
//  Pioneer
//
//  Created by Beh on 2021/5/22.
//

import Foundation
public class Device : NSObject {
    /// 设备注册时间戳
    var createdAt: Date
    /// 最后一次属性更新时间
    var lastUpdatedAt: Date
    
    /// 临时设备标识符
    public var identifier: TargetIdentifier
    
    init(_ identifier: TargetIdentifier) {
        self.createdAt = Date()
        self.identifier = identifier
        lastUpdatedAt = createdAt
    }
}
